### ISSUE TITLE
terraform_unused_declarations false negatives

### ISSUE THREAD

**Issue by weberc2-tempus** [terraform-linters/tflint#1122]
The `terraform_unused_declarations` rule seems broken on even the most basic conceivable example, so I assume I'm doing something wrong, but it's not clear what that might be.

## Reproducible Code

```hcl
locals {
  foo = "foo"
}

module "module" {
  source = "../module"
  foo    = locals.foo
}
```

```console
$ tflint --enable-rule=terraform_unused_declarations
1 issue(s) found:

Warning: local.foo is declared but not used (terraform_unused_declarations)

  on foo.tf line 2:
   2:   foo = "foo"

Reference: https://github.com/terraform-linters/tflint/blob/v0.28.1/docs/rules/terraform_unused_declarations.md
```

## Version

```console
$ tflint -v
TFLint version 0.28.1
$ terraform -v
Terraform v0.13.5

Your version of Terraform is out of date! The latest version
is 0.15.3. You can update by downloading from https://www.terraform.io/downloads.html
```

## Trace Output

```console
$ TFLINT_LOG=trace tflint --enable-rule=terraform_unused_declarations
12:26:43 config.go:96: [INFO] Load config: .tflint.hcl
12:26:43 config.go:108: [INFO] Default config file is not found. Ignored
12:26:43 config.go:117: [INFO] Load fallback config: /Users/.../.tflint.hcl
12:26:43 config.go:125: [INFO] Fallback config file is not found. Ignored
12:26:43 config.go:127: [INFO] Use default config
12:26:43 option.go:49: [DEBUG] CLI Options
12:26:43 option.go:50: [DEBUG]   Module: false
12:26:43 option.go:51: [DEBUG]   Force: false
12:26:43 option.go:52: [DEBUG]   IgnoreModules: map[string]bool{}
12:26:43 option.go:53: [DEBUG]   EnableRules: []string{"terraform_unused_declarations"}
12:26:43 option.go:54: [DEBUG]   DisableRules: []string(nil)
12:26:43 option.go:55: [DEBUG]   Only: []string(nil)
12:26:43 option.go:56: [DEBUG]   EnablePlugins: []string(nil)
12:26:43 option.go:57: [DEBUG]   Varfiles: []string{}
12:26:43 option.go:58: [DEBUG]   Variables: []string{}
12:26:43 loader.go:57: [INFO] Initialize new loader
12:26:43 loader.go:82: [INFO] Load configurations under .
12:26:43 loader.go:90: [INFO] Module inspection is disabled. Building a root module without children...
12:26:43 loader.go:170: [INFO] Load values files
12:26:43 runner.go:50: [INFO] Initialize new runner for root
12:26:43 provider.go:62: [INFO] Prepare rules
12:26:43 provider.go:74: [DEBUG] `terraform_unused_declarations` is enabled
12:26:43 provider.go:90: [INFO]   4 default rules enabled
12:26:43 terraform_deprecated_interpolation.go:48: [TRACE] Check `terraform_deprecated_interpolation` rule for `root` runner
12:26:43 terraform_module_pinned_source.go:72: [TRACE] Check `terraform_module_pinned_source` rule for `root` runner
12:26:43 terraform_module_pinned_source.go:81: [DEBUG] Walk `module.source` attribute
12:26:43 terraform_workspace_remote.go:48: [TRACE] Check `terraform_workspace_remote` rule for `root` runner
12:26:43 terraform_unused_declaration.go:55: [TRACE] Check `terraform_unused_declarations` rule for `root` runner
1 issue(s) found:

Warning: local.foo is declared but not used (terraform_unused_declarations)

  on foo.tf line 2:
   2:   foo = "foo"

Reference: https://github.com/terraform-linters/tflint/blob/v0.28.1/docs/rules/terraform_unused_declarations.md
```

**Comment by weberc2-tempus** [comment]
Oof, never mind. A colleague finally figured out that the issue was that I was using `locals.foo` instead of `local.foo`. 🤦‍♂️

**Comment by bendrucker** [comment]
> broken on even the most basic conceivable example

Check your tone please. 

In the future, checking out the tests should help show you the correct usage. And if there is a legitimate bug, a failing test case is a significant contribution towards fixing.
